src/catchup/CatchupManagerImpl.cpp

```diff
@@ -450,16 +511,47 @@ CatchupManagerImpl::tryApplySyncingLedgers()
         auto const& lcd = it->second;
 
         // we still have a missing ledger
-        if (ledgerHeader.header.ledgerSeq + 1 != lcd.getLedgerSeq())
+        if (nextToClose != lcd.getLedgerSeq())
+        {
+            break;
+        }
+
+        // If we have too many ledgers queued to apply, just stop scheduling
+        // more and let the node gracefully go into catchup.
+        releaseAssert(mLastQueuedToApply >= lcl);
+        if (nextToClose - lcl >= MAX_EXTERNALIZE_LEDGER_APPLY_DRIFT)
```

Comment:
Is this a condition we observe in reality? Or, I mean, why is there new code here to handle it? Is it something you're concerned will happen due to speed mismatches between the two threads?

Reply:
So the scenario I'm thinking of is if node is externalizing ledgers much faster than it can apply them. This may create a weird situation where ledger close queue grows without bounds and node technically reports as being "in sync" whereas in reality it's arbitrarily behind. This is also not great for flooding: while it'll technically "work", the node will likely flood obsolete/irrelevant traffic. So with this change we allow nodes to fall behind by about a minute of consensus, and if they can't keep up, we stop scheduling new ledgers to apply, and let core go into catchup (it will wait until all queued ledgers are applied first, then start catchup after trigger ledger)